pump/storage/storage.go

```diff
@@ -660,6 +660,12 @@ func (a *Append) doGCTS(ts int64) {
 
 	deleteNum := 0
 
+	irange := &util.Range{
+		Start: encodeTSKey(0),
+		Limit: encodeTSKey(ts + 1),
+	}
+	iter := a.metadata.NewIterator(irange, nil)
```

Comment:
In the original implementation, the `iter` is created and released every time in the loop to scan all the remaining records in the range `[0, ts]`.
After this change, we'll be using the `iter` after releasing it, which seems buggy?
To trigger the bug, we'll need to GC at least `1024 * 100` records, which explains why the tests were passing.